sonar-batch/src/main/java/org/sonar/batch/bootstrap/BatchComponents.java

```diff
@@ -66,8 +67,16 @@ private BatchComponents() {
       HtmlReport.class,
       IssuesReportBuilder.class,
       SourceProvider.class,
-      RuleNameProvider.class
-      );
+      RuleNameProvider.class,
+
+      // Tasks
+      Tasks.class,
+      ListTask.DEFINITION,
+      ListTask.class,
+      ScanTask.DEFINITION,
+      ScanTask.class,
+      ViewsTask.DEFINITION,
+      ViewsTask.class);
```

Comment:
I have just added one more medium test that demonstrates that pico is lazy. That's indeed strange because we are using ReflectionLifecycleStrategy that returns "false" for "isLazy"... And regarding failing fast I usually agree but in the past people were complaining that for example a bug in views task was preventing them to run analysis.